src/collections/map.rs

```diff
@@ -201,17 +280,16 @@ where
     }
 }
 
-fn encode_bound<K>(bound: Bound<&K>) -> Result<Bound<Vec<u8>>>
-where
-    K: Encode,
-{
+fn encode_bound<K: Encode + Clone>(bound: Bound<&K>) -> Result<Bound<Vec<u8>>> {
     match bound {
         Bound::Included(inner) => {
-            let encoded_bound = Encode::encode(inner)?;
+            let map_key = MapKey::<K>::new(inner.clone())?;
```

Comment:
What's `MapKey` doing here, wouldn't this be equivalent to just doing `inner.encode()`?

Reply:
Good catch. Changing this let me remove the Clone bound here as well.